src/cplus_plugin/gui/constant_raster_manager_dialog.py

```diff
@@ -0,0 +1,1135 @@
+            if items_updated:
+                msg_parts.append(f"Updated {len(items_updated)} component(s)")
+            msg = " and ".join(msg_parts)
+
+            if current_value is not None:
+                self.message_bar.pushInfo(
+                    "Components Configured", f"{msg} with value: {current_value}"
+                )
+            else:
+                self.message_bar.pushWarning(
+                    "Using Default Values",
+                    f"{msg} with default value (0.0). Enter a value in the widget first.",
+                )
+
+        # NOTE: We don't call normalize() here anymore because:
+        # 1. The user specifies min/max values explicitly in the UI
+        # 2. Auto-normalizing would override user values
+        # 3. With only one component, min==max causes incorrect 0.5 fallback
+
+    def _get_pixel_size_from_snap_layer(self) -> float:
+        """Get pixel size from the snap layer for consistency with analysis.
+
+        The snap layer is used as a reference to align all rasters during
+        scenario analysis. Using its pixel size ensures constant rasters
+        are consistent with other processing outputs.
+
+        :returns: Pixel size in map units, defaults to 30.0 if snap layer unavailable
+        """
+        DEFAULT_PIXEL_SIZE = 30.0
+
+        # Get snap layer path from settings
+        snap_layer_path = settings_manager.get_value(Settings.SNAP_LAYER, default="")
+
+        if not snap_layer_path:
+            return DEFAULT_PIXEL_SIZE
+
+        if not os.path.exists(snap_layer_path):
+            return DEFAULT_PIXEL_SIZE
+
+        try:
+            snap_layer = QgsRasterLayer(snap_layer_path, "snap", "gdal")
+            if not snap_layer.isValid():
+                log(
+                    f"Invalid snap layer: {snap_layer_path}, using default pixel size",
+                    info=False,
+                )
+                return DEFAULT_PIXEL_SIZE
+
+            # Get pixel size (use X resolution, assuming square pixels)
+            pixel_size_x = snap_layer.rasterUnitsPerPixelX()
+            pixel_size_y = snap_layer.rasterUnitsPerPixelY()
+
+            # Use average if X and Y differ
+            pixel_size = (pixel_size_x + pixel_size_y) / 2.0
+            return pixel_size
+
+        except Exception as e:
+            log(
+                f"Error reading snap layer pixel size: {str(e)}, using default",
+                info=False,
+            )
+            return DEFAULT_PIXEL_SIZE
+
+    def _create_context(self) -> ConstantRasterContext:
+        """Create a ConstantRasterContext from current map canvas and settings.
+
+        :returns: Configured ConstantRasterContext
+        """
+        # Try to get extent from AOI/Study Area first
+        studyarea_path = settings_manager.get_value(Settings.STUDYAREA_PATH)
+        use_aoi = studyarea_path and os.path.exists(studyarea_path)
+
+        if use_aoi:
```

Comment:
@osundwajeff I think there is a flag that defines the option that the user defined in step 1. That should specify the path we take here i.e. layer or bounding box.